### ISSUE TITLE
What's the status on Google Cast implementation?

### ISSUE THREAD

**Issue by ale-reyes** [microg/android_packages_apps_GmsCore#580]
I checked the wiki and the implementation page hasn't been updated in two years, so I thought I'd ask.

**Comment by benwaffle** [comment]
#79

**Comment by cubagithub** [comment]
#79  is Closed.
Can anyone use cast function natively? I can't use it on Android 11 + MicroG using a FireTV or Chromecast.
![signal-2021-03-02-183150_002](https://user-images.githubusercontent.com/34730232/109613824-d3403780-7b85-11eb-84e7-91a819edeb6d.png)
![signal-2021-03-02-183150_003](https://user-images.githubusercontent.com/34730232/109613828-d3d8ce00-7b85-11eb-8729-4f1ae385fbff.png)
![signal-2021-03-02-183150_001](https://user-images.githubusercontent.com/34730232/109613829-d4716480-7b85-11eb-9b0f-d61e57636ba0.png)

**Comment by ale5000-git** [comment]
Fortunately on some ROM the removal is reverted.

**Comment by hollowshiroyuki** [comment]
Hi, I have the same issue using the unofficial Lineage MicroG on a Pixel 3 :
Lineage Version : 19-20220622-microG-blueline (Android 12)
MicroG Version : 0.2.24.214816

Is https://github.com/TeamVanced/Vanced/issues/688#issuecomment-992102591 relevant ?

The chromecast was tested with an unmodified Xiaomi Note 10 too and worked correctly.

**Comment by matteopessina** [comment]
Any news on this feature?

I did not managed to cast Netflix, Display+ directly to chromecast.
I am using /e/OS 1.6-202212208238947, with microg 0.2.25.223616-9 on OnePlus8 IN2013.

I tried debugging the Google sample cast sender application https://github.com/googlecast/CastVideos-android.
On an old Huawei P10 with stock rom it works, while on os with microg it shows the following logcat
[cast-logcat.txt](https://github.com/microg/GmsCore/files/10322469/cast-logcat.txt)

The most interesting part is the following:
```
/GmsDynamiteLoaderImpl: No such module known: com.google.android.gms.cast.framework.dynamite
/GmsDynamiteLoaderImpl: returning temp fix module version for com.google.android.gms.cast.framework.dynamite. Cast API wil not be functional!
```

**Comment by mattszcz** [comment]
The following information seems like it holds the clue to solving the Chromecast casting issue:

>With old versions of YT Vanced (15.43.32) and Vanced Micro G (0.2.22.212658) I'm able to cast to my non-smart TV using chromecast. So I don't get why using chromecast would not work with revanced (tried with a newer version of YT and the same 0.2.22.212658 Micro G version).

Source: https://www.reddit.com/r/revancedapp/comments/wi4whv/casting_to_chromecast/iosvuag

**Comment by benjamingwynn** [comment]
The Gnome team recently got Chromecast working, perhaps something useful could be found in this PR? https://gitlab.gnome.org/GNOME/gnome-network-displays/-/merge_requests/171

**Comment by bertin0** [comment]
> The Gnome team recently got Chromecast working, perhaps something useful could be found in this PR? https://gitlab.gnome.org/GNOME/gnome-network-displays/-/merge_requests/171

Seems interesting but i think that they only implemented the functionality  to cast a stream from the device to the chromecast (like sharing your screen), not directly from the internet to the chromecast. I just realised that my rom also has this functionality implemented, but I'm not sure whether that goes through the chromecast or the miracast/wfd api.

**Comment by minimoonleaf** [comment]
> 以下信息似乎包含解決 Chromecast 投射問題的線索：
> 
> > 使用舊版本的 YT Vanced (15.43.32) 和 Vanced Micro G (0.2.22.212658)，我可以使用 chromecast 投射到我的非智能電視。所以我不明白為什麼使用 chromecast 不能與 revanced 一起使用（嘗試使用較新版本的 YT 和相同的 0.2.22.212658 Micro G 版本）。
> 
> 來源：[https ://www.reddit.com/r/revancedapp/comments/wi4whv/casting_to_chromecast/iosvuag](https://www.reddit.com/r/revancedapp/comments/wi4whv/casting_to_chromecast/iosvuag)

The older version can support MicroG. Why the new MicroG can't support chromecast ? I'm also curious to know why it can't support Chromecast?

**Comment by bertin0** [comment]
> > 以下信息似乎包含解決 Chromecast 投射問題的線索：
> > > 使用舊版本的 YT Vanced (15.43.32) 和 Vanced Micro G (0.2.22.212658)，我可以使用 chromecast 投射到我的非智能電視。所以我不明白為什麼使用 chromecast 不能與 revanced 一起使用（嘗試使用較新版本的 YT 和相同的 0.2.22.212658 Micro G 版本）。
> > 
> > 
> > 來源：[https ://www.reddit.com/r/revancedapp/comments/wi4whv/casting_to_chromecast/iosvuag](https://www.reddit.com/r/revancedapp/comments/wi4whv/casting_to_chromecast/iosvuag)
> 
> The older version can support MicroG. Why the new MicroG can't support chromecast ? I'm also curious to know why it can't support Chromecast?

Because there is the old Chromecast API v1 and the newer V2. Microg has v1 implemented but not V2. The old vanced version uses the old Chromecast v1 API, while the newer revanced uses V2. Ig one way for the revanced devs to fix this would be to force the newer app to use the older API but I'm not sure how feasible that is. Also it's not within the scope for microg devs.

**Comment by minimoonleaf** [comment]
> Because there is the old Chromecast API v1 and the newer V2. Microg has v1 implemented but not V2. The old vanced version uses the old Chromecast v1 API, while the newer revanced uses V2. Ig one way for the revanced devs to fix this would be to force the newer app to use the older API but I'm not sure how feasible that is. Also it's not within the scope for microg devs.

So is there no one developing chromecast V2 API applications now? Why is ReVanced completely unable to use chromecast now? If there are still APIs that support V1 or V2, shouldn’t the function be normal?

**Comment by trmdi** [comment]
May I ask an unrelated question: Why can't my phone find Youtube on my Android Tv Box? Does it still use the DIAL protocol or another one?

**Comment by Quanalogy** [comment]
> > > 以下信息似乎包含解決 Chromecast 投射問題的線索：
> > > > 使用舊版本的 YT Vanced (15.43.32) 和 Vanced Micro G (0.2.22.212658)，我可以使用 chromecast 投射到我的非智能電視。所以我不明白為什麼使用 chromecast 不能與 revanced 一起使用（嘗試使用較新版本的 YT 和相同的 0.2.22.212658 Micro G 版本）。
> > > 
> > > 
> > > 來源：[https ://www.reddit.com/r/revancedapp/comments/wi4whv/casting_to_chromecast/iosvuag](https://www.reddit.com/r/revancedapp/comments/wi4whv/casting_to_chromecast/iosvuag)
> > 
> > 
> > The older version can support MicroG. Why the new MicroG can't support chromecast ? I'm also curious to know why it can't support Chromecast?
> 
> Because there is the old Chromecast API v1 and the newer V2. Microg has v1 implemented but not V2. The old vanced version uses the old Chromecast v1 API, while the newer revanced uses V2. Ig one way for the revanced devs to fix this would be to force the newer app to use the older API but I'm not sure how feasible that is. Also it's not within the scope for microg devs.

Do you have any insights into how big the assignment of implementing the v2 api is in the current state? 
I'm wondering if there might be any blockers outside simply supporting the api? 
Do you know if it is a gigantic api that is expected years to develop or what might be the reason for this v2 api not being implemented yet?

Might there already be some development on this or does it need to be done from scratch?

**Comment by Tthecreator** [comment]
Hi, 

How is everyone doing.
I'm using MicroG, and get frequent complaints from my Girlfriend if she has to turn on the casting, instead of me.

I will see if I can take a look at this.

Currently, I have a plan to take some of the code from VLC (which has a complete open source implementation) and see if that somehow can be used or translated for MicroG. First, I'll be starting off seeing if I can get mDNS to work. Then, see if and how I could start a stream.

I will also hope to not run into license issues.

**Comment by Tthecreator** [comment]
Update:

Okay, seems like I got a know-how of the Chromecast protocol. That seems easy enough right now.
And I didn't need VLC after all.

Anyways, the really big challenge for me at this moment is to understand how this will integrate into microg. The problem is that (I think) the cast-framework is linked into the application and still black-box proprietary. Meanwhile, our implementation in the cast core, doesn´t seem to work. But I can't tell what the communication should look like. 
In CastMediaRouteController, there is an onSelect() function that is not implemented. I cannot see any other calls to our framework concerning casting. I therefore suspect that this call needs to trigger another call or callback somehow somewhere in order for the framework and app code to continue making the connection. But it is in no way clear to me what that is. I've tried to just setup the connection to the chromecast. I can see communication happen (I have a MITM setup), but none that are specific to my youtube app that I'm testing with. Also, in Youtube, the cast button works just fine, but in other apps it either remains grayed-out or invisible. Another strange behavior I don't know why it's doing that.

So wish me luck. And if anyone is available to help me, that would also be great.

**Comment by Tthecreator** [comment]
@matteopessina 
It seems that Adam Mills or @emlove may have more knowledge on the matter as well. Perhaps they can tune in.

**Comment by mar-v-in** [comment]
Hi. I haven't really worked on Cast and don't know a lot about it.

**Comment by Tthecreator** [comment]
Hi @AnyMelz At the time, I had found the same information. But I was still puzzled. 
The problem is that there are a few different layers of communication:
Cast device <--TCP--> Play services framework <-- Some IPC?? --> Play services library (linked in App I think) <-- Java function calls --> An actual App.

The framework takes over a lot of the responsibility from the app. See: https://developers.google.com/cast/docs/android_sender/integrate#app_flow.

It has been made really opaque to me what exactly is happening, or what is supposed to happen. What communication is supposed to go where? It's not like Google provides and extensive sequence diagram of what happens (except for the app to lib part maybe.) That makes things really hard to trace. So in the end back then after spending almost a week of my summer holiday on this, I gave up. I could try again sometime when I feel confident again. Maybe I'm just missing an 'eureka moment'.

**Comment by Tthecreator** [comment]
Does anyone know what this means:

`Service not supported: GetServiceRequest{serviceId=UNKNOWN(161), gmsVersion=232800000, packageName='com.google.android.youtube', extras=Bundle[EMPTY_PARCEL]}`

**Comment by Solain** [comment]
Any updates guys? Or any alternative ways to cast to tv?

**Comment by ThaChillera** [comment]
@Solain For what its worth I've very recently discovered after upgrading to LineageOS for MicroG 21 (Android 14) on my Samsung S10, Spotify casting to TV works very well. Casting Spotify to Audiocast doesn't work at all, sadly

**Comment by Smart123s** [comment]
> Spotify casting to TV works very

Spotify has its own casting solution called Spotify Connect that doesn't use Chromecast. Are you sure you didn't use that function? The easiest way to test Chromecast is with the YouTube app (without TV code).

**Comment by Solain** [comment]
@ThaChillera yeah spotify has its own mechanism, its not google/chrome cast. They use upnp or some other thing. Thank you in any way <3

**Comment by ehem** [comment]
I guess I have to join the chorus of people waiting for V2 cast protocol support.  Alas my neurons are targeted elsewhere right now.

I offer US$500 to anyone who manages to complete this task before the end 2025. 
 The favored online transfer method is PayPal, it could take 4 months for a transfer if this occurs.  This is not to say I wouldn't offer this after 2025, only that the present offer is not valid forever.

**Comment by Tthecreator** [comment]
@ehem I'm going to be straight with you. I've invested significant hours on this problem for two occasions at this point.
Both times I just couldn't wrap my head around how the framework exactly works and what parts are functioning how.
I cannot who mentioned it, or where it was mentioned, but someone said that it's likely not the core cast API itself, but some integrity/security/validation thing involved with it. But I forgot what this was called.

Anyways, for now I'm also hoping that someone that actually works often with Android and Java (not me, I do embedded) and is more familiar with all the coding patterns from that world has more luck with this implementation. Just as much as you do...

**Comment by ehem** [comment]
You're not the only one who tends to be closer to the OS/hardware and thinks Java/Android do a number of things in odd ways.  I'm guessing a bounty might be the way to go.

Looking at things I could almost afford to hire someone to get V2 implemented, though I lack anyone with appropriate knowledge.

**Comment by ehem** [comment]
Any chance the [wiki](https://github.com/microg/GmsCore/wiki/Implementation-Status) could be updated in light of this?  The "Issues" column being "Minor issues" isn't very informative.  Instead it should say "Cast V1 protocol implemented, Cast V2 protocol unimplemented; essentially unusable".

**Comment by mattszcz** [comment]
Surely AI could solve whatever coding problem we are having here.

**Comment by lucasmz-dev** [comment]
@ehem 
> I offer US$500 to anyone who manages to complete this task before the end 2025.

Some people seem to be using bountyhub.dev for this stuff. Unsure of its trustworthiness though. Guess we'll know soon with the two pinned issues.

**Comment by omarsoufiane** [comment]
> [@ehem](https://github.com/ehem)
> 
> > I offer US$500 to anyone who manages to complete this task before the end 2025.
> 
> Some people seem to be using bountyhub.dev for this stuff. Unsure of its trustworthiness though. Guess we'll know soon with the two pinned issues. 


I'm The founder of bountyhub. I do understand your worries about trustworthiness  especially since we are a relatively new platform (just over a year old). but now we do support promised bounties where we don't hold funds and you only pay once the bounty is solved. We have paid out many bounties up $1000 on other projects, but our main issue is that unlike other projects we have a 0% success rate with microg.
There are 4 bounties from microg (including this one) on bountyhub and not a single valid pull request has been submitted yet. 
maybe we should do more ads,  or maybe it's because this project looks intimidating for first time contributors. I have been coding for more than 10 years, and I find it hard to setup a testing environment on an emulator. especially on some issues that require specific hardware like this one. 
But we may never know, some devs just need a small incentive to spend more time on open source. And I hope we can improve our success rate this time.

**Comment by Tthecreator** [comment]
@omarsoufiane I think the right person is someone that is in the Java world and has a native understanding of all the design patterns used everywhere. When having making changes to a running code base, you can learn how these things work because you can see how everything is connected. But when you introduce a whole black box that you are trying to re-implement, the learning curve is simply to steep. If you are not used to the paradigms, the code is too much spaghetti and boilerplate, things happening across several libraries and system components, often async. I hope someone that is more involved in technical Java Android development in their day job could give a hand here.

I appreciate the bounty by emeitner and I hope it will result in success!

**Comment by Excellencedev** [comment]
We can also use https://algora.io to put bounties

**Comment by omarsoufiane** [comment]
@Excellencedev algora doesn't support crowdfunding where each individual contributor can decide whether to reward a bounty or not, it's more for companies who own open source projects and want to hire developpers. it's useful especially when you want to pick a developer and pay them him by hour.

**Comment by energypantry** [comment]
Hi, I reviewed this bounty and I can take it.

Plan:
1) Reproduce/confirm acceptance criteria
2) Implement in small milestones with reversible changes
3) Submit verification evidence (tests/logs/screenshots)

ETA: first technical update within 24 hours.

[zhi-outreach-r3:microg_GmsCore#580]

**Comment by Excellencedev** [comment]
> Hi, I reviewed this bounty and I can take it.
> 
> Plan:
> 
> 1. Reproduce/confirm acceptance criteria
> 2. Implement in small milestones with reversible changes
> 3. Submit verification evidence (tests/logs/screenshots)
> 
> ETA: first technical update within 24 hours.
> 
> [zhi-outreach-r3:microg_GmsCore#580]

This guy is a bot. I don't think he's a human. He sent bounty claim in multiple other bounty issues at almost the same time

**Comment by energypantry** [comment]
I can move this from discussion to executable scope.

Next deliverable: by **2026-02-17 12:00 UTC** I will post an implementation brief (affected modules, milestones, test plan) for Google Cast support in microG, followed by a PoC execution path.

Pricing options:
- **A) Quick**: feasibility + architecture brief — **$120 or 18% bounty**
- **B) Standard**: architecture + PoC patch for discovery/session flow — **$260 or 24% bounty**
- **C) Urgent**: end-to-end PoC + test logs + rollout roadmap — **$480 or 32% bounty**

CTA: reply with **A/B/C** and acceptance criteria:
1) required cast scenarios/devices,
2) minimum Android versions,
3) Done = PoC evidence or production-ready patch.

**Comment by mattszcz** [comment]
> This guy is a bot. I don't think he's a human. He sent bounty claim in multiple other bounty issues at almost the same time

As long as the work gets done. Let the bots do it.

**Comment by energypantry** [comment]
@ale5000-git quick conversion proposal:

24h verifiable deliverable:
- By `2026-02-17T03:30:00Z` I will publish a Draft PR with Cast handshake trace capture, one minimal connection-path patch, and a reproducible verification checklist for maintainers.

Pricing (A/B/C):
- A) $220 (minimal scope, single-pass)
- B) $460 (standard scope, full acceptance checklist)
- C) $820 (priority turnaround + same-day iteration window)

Acceptance CTA:
Reply `A`, `B`, or `C` and confirm your acceptance criteria + preferred UTC start time. I will start immediately after your confirmation and post the first checkpoint in-thread.

**Comment by Excellencedev** [comment]
> > This guy is a bot. I don't think he's a human. He sent bounty claim in multiple other bounty issues at almost the same time
> 
> As long as the work gets done. Let the bots do it.

Actually once one bot starts coming. Much more start to come. The only target bounties. Only bounties
Heres a good example: https://github.com/coollabsio/coolify/issues/8154
IMHO, I dont think they can solve this

**Comment by Adityapradh** [comment]
I would like to work on this issue.

**Comment by Shanu-Kumawat** [comment]
Hey folks, I saw the bounty on this and I'm looking into picking it up. I know implementing something like Google Cast is a massive undertaking, but I've been doing a lot of deep systems level work recently (had a few PRs merged into the core Dart SDK analyzer/ffi), so I am confident enough to get this done.

Just want to do a check on two things before I lose my weekend setting up test environments:

1. Are the maintainers actually open to reviewing a PR for this right now? I know a feature this big takes a ton of maintainer bandwidth to review, so I just want to make sure the door is open before I start.
2. Are the bounties still active?

If yes to both, I'll start working on the codebase this week!

**Comment by hasali19** [comment]
I've been looking into this recently to try and get casting from youtube revanced working. While I haven't had much luck with revanced yet, I have managed to get casting from the official youtube app to work. It's still a bit buggy, particularly around the disconnect process, but otherwise mostly seems to be functional. I've put up a [PR](https://github.com/microg/GmsCore/pull/3377) with what I have so far if anyone wants to try it out.

**Comment by ehem** [comment]
[My bounty](https://github.com/microg/GmsCore/issues/580#issuecomment-2479755158) had definitely expired.  Since I do still want this feature I do hereby extend the bounty until the end of 2026.  Thanks for asking as expiration of bounties does happen and I wouldn't want you to be disappointed.

**Comment by OsirisLord** [comment]
I've submitted PR #3505 that implements the stubbed `CastMediaRouteController` methods. This is the reason the cast button appears but selecting a device does nothing — `onSelect()`, `onControlRequest()`, `onRelease()` were all no-ops.

What this PR does:
- Connects to the Chromecast device on `onSelect()` using the existing `su.litvak.chromecast` library dependency (already handles TLS/auth)
- Dispatches `MediaControlIntent` actions (play/pause/seek/stop/session management) to Chromecast protocol commands
- Properly disconnects on `onRelease()`/`onUnselect()`
- Handles volume control

The scope is limited to the MediaRouteProvider → RouteController pathway. The Cast v2 framework dynamite module is a separate issue that will need further work.

**Comment by Jorel97** [comment]
I opened #3554 as a small, reviewable supporting patch for the missing `com.google.android.gms.cast.framework.dynamite` descriptor that appears in the logs discussed here.

Before I spend more time on a larger Cast v2 change, I want to keep the direction realistic and reviewable:

- avoid an opaque `CastMediaRouteController` rewrite like the concerns raised around #3505;
- inspect the partial working path from #3377, especially the framework/session/controller wiring that reportedly gets official YouTube mostly working;
- continue in smaller PRs only if that direction is useful to maintainers.

Maintainer or contributor guidance would help a lot: should follow-up work be split into small framework/session patches, based on/rebased from #3377, or held until someone can provide device-level verification expectations? I can continue, but I do not want to add noise to this issue with a large untestable patch.

**Comment by peterhel** [comment]
**tl;dr:** the thing that makes modern Cast apps (Prime, YouTube, Netflix) silently fail on microG isn't the classic Cast path — it's that they drive the device over the **connectionless** API (`Cast.API_CXLESS`), which microG doesn't implement. That gap is now well understood and has been validated end-to-end on real hardware, and several open PRs are converging on it. I've spent a while building and hardware-testing each of them, so here's a map of where things stand, in case it helps us land one.

Full disclosure up front: two of the PRs below (#3567, #3570) are mine, so weight my take accordingly — I've tried to keep this factual (build results + RE) rather than advocacy.

### Why casting silently fails today
microG implements the classic, connection-based `CastDeviceController`. But the modern Cast SDK — and every first-party app I've tested — drives the device over the **connectionless** path:
- bind the CAST service with the `cxless_*` features,
- deliver the listener out-of-band via `setListener`/`addListener` (not via the bind `extras`),
- call `connect()`, then **wait for the service to call the listener's connected-callback with `0`** before launching anything.

microG never implements that handshake, so the session binds and then stalls forever. (Withholding the `cxless_*` features to force the classic path instead just makes the SDK declare microG too old — `ConnectionResult=2`.) The service also has to advertise the requested `apiFeatures` via `onPostInitCompleteWithConnectionInfo`, or you hit the same version error.

### The wire contract (reverse-engineered + verified)
From `play-services-cast` 21.4.0, cross-checked by building it and running the **real Prime Video app** against a Chromecast — it launches Prime's own DRM receiver and plays a title from a de-Googled phone:
- `ICastDeviceController`: `connect() = 16`, `setListener`/`addListener = 17`, `unregisterListener = 18`
- `ICastDeviceControllerListener`: readiness callback is **`onConnectedWithResult(int statusCode) = 13`** (wire txn 14) — service emits `0` after `connect()`
- AIDL dispatches by transaction id, so the **id + arg types are the contract; method names are cosmetic**
- `chromecast.connect()` (the su.litvak fork) throws `GeneralSecurityException` as well as `IOException` — catching only `IOException` won't compile

Encouragingly, #3377 independently reverse-engineered the same contract (`connect=16`, `addListener=17`, listener connected-callback as an `int` at `=13`). Two independent passes landing on the same shape is good evidence it's right.

### Where the open PRs stand (I built each)
- **#3567 / #3570** (mine — session-start trigger + connectionless device controller): build, validated end-to-end against real Prime.
- **#3377** (connectionless, YouTube-focused): builds; correct `int` listener signature; one missing null-check (NPE risk) noted on the PR.
- **#3502** (connectionless, comprehensive incl. framework): doesn't compile as-is (`GeneralSecurityException` unhandled + some framework type errors), and uses `onConnected(String)` at the listener slot — the one signature that diverges from the above.
- **#3470** (robustness: route socket ops off the main thread, correct `GeneralSecurityException` lifecycle, `joinApplication`): builds; complementary to the connectionless work, touches the same file.
- **#3505** (implements `CastMediaRouteController`): the file landed base64-encoded so it doesn't compile; the decoded source is the legacy `onControlRequest`/RemotePlaybackClient path, which is separate from the connectionless path modern apps use.
- **#3554** (dynamite `ModuleDescriptor`): builds; clears the "Cast API wil not be functional" temp-fix warning. Independent and safe.

### The frustrating part — and the opportunity
There's a lot of overlapping effort (#3502/#3570/#3377 all reimplement the connectionless device controller on the same files), so they collide and none has merged. But the pieces to land a working Cast are arguably *all present* across these PRs: the connectionless handshake on the now-converged contract, the `GeneralSecurityException`-correct lifecycle from #3470, the dynamite descriptor (#3554), and the session-start trigger (#3567).

One honest caveat: there's also an auxiliary `CAST_API` service (the client binds it; microG logs "service not supported") that's still unimplemented — but it didn't block media playback in my testing, so the connectionless device controller is the load-bearing piece.

I'm happy to help reconcile these into one mergeable device controller and to hardware-test candidates — I have the rig and the RE notes. Whatever shape the maintainers prefer; mainly flagging that the core protocol is now nailed down and proven, so this feels close.

**Comment by aquette** [comment]
Any update @peterhel ?
Your work sounds really promising!

**Comment by waterWang** [comment]
I've submitted a PR (#3668) that implements Google Cast functionality by implementing the previously stubbed CastMediaRouteController methods. The PR is ready for review — it enables selecting Chromecast devices, media control (play/pause/seek/stop), and volume control. Could a maintainer please take a look?

**Comment by peterhel** [comment]
Thanks @aquette, that means a lot! Not much new since the writeup above: my connectionless PRs still build and still cast real apps on my rig, so the protocol's solid. Mostly it needs a maintainer to pick which of the overlapping PRs to build on so they can be merged into one. Happy to keep hardware-testing whenever that happens.

**Comment by chenel** [comment]
Any idea which maintainer(s) need to be pestered?  This is the last remaining annoyance I have with Murena eOS on the Fairphone, and for it to be so close and  still languishing is galling!

**Comment by DonnieWest** [comment]
Has anyone built any of these solutions and vetted that they work? If someone can vouch for em, might as well ping Marvin but I wouldn't until it's been vetted

**Comment by xvello** [comment]
> Any idea which maintainer(s) need to be pestered?

@chenel please DO NOT pester volunteer maintainers. They will engage on their own time.
I'm also frustrated that it's not implemented, but I'm frustrated at Google for deprecating the v1 protocol, not at maintainers who might be reluctant to release a reverse-engineered implementation: there's a high likelihood that many edge cases will need to be ironed out, and they might not have the energy and time to handle that support load.
